src/pytest_memray/marks.py

```diff
@@ -186,15 +186,27 @@ def _passes_filter(
 def limit_memory(
     limit: str,
     *,
+    current_thread_only: bool = False,
     _result_file: Path,
     _config: Config,
     _test_id: str,
 ) -> _MemoryInfo | _MoreMemoryInfo | None:
     """Limit memory used by the test."""
     reader = FileReader(_result_file)
-    allocations: list[AllocationRecord] = list(
-        reader.get_high_watermark_allocation_records(merge_threads=True)
-    )
+    allocations: list[AllocationRecord] = []
+    if current_thread_only:
+        main_thread = reader.metadata.main_thread_id
+        allocations.extend(
+            record
+            for record in reader.get_high_watermark_allocation_records(
+                merge_threads=False
+            )
+            if record.tid == main_thread
+        )
+    else:
+        allocations.extend(
+            reader.get_high_watermark_allocation_records(merge_threads=True)
+        )
```

Comment:
Why not go for the more obvious:
```suggestion
    allocations: list[AllocationRecord] = [
        record
        for record in reader.get_high_watermark_allocation_records(
            merge_threads=not current_thread_only
        )
        if not current_thread_only or record.tid == reader.metadata.main_thread_id
    ]
```